### ISSUE TITLE
Include mesh location in cube printout

### ISSUE THREAD

**Issue by wjbenfold** [SciTools/iris#4756]
## ✨ Feature Request

In the cube printout, include the mesh location.

## Motivation

This important information is currently absent

**Comment by pp-mo** [comment]
I think it also makes sense to print "the mesh"
 -- i.e. its name, which is usually a valid thing, certainly when loaded from a file, it at least has a var_name
The ugrid-checker [prints this for the netcdf data variables](https://github.com/pp-mo/ugrid-checks#command-line--structure-analysis) (~the cubes, in Iris), like this :
```
Mesh Data Variables
    "sample_data"
        mesh : "topology"
        location : "face"
```

So, I think the key question is _where+how_ should this info appear in the cube listing.
As a starting suggestion , what currently looks like ...
```
>>> print(meshcube)
values / (m)                        (model_level: 2; -- : 3)
    Dimension coordinates:
        model_level                             x       -
    Mesh coordinates:
        latitude                                -       x
        longitude                               -       x
    Auxiliary coordinates:
        aux                                     x       x
    Attributes:
        extra                       3.2

```

Might now look like ...
```
>>> print(meshcube)
values / (m)                        (model_level: 2; -- : 3)
    Dimension coordinates:
        model_level                             x       -
    Mesh:
        Mesh2d : face                                   x
    Mesh coordinates:
        latitude                                -       x
        longitude                               -       x
    Auxiliary coordinates:
        aux                                     x       x
    Attributes:
        extra                       3.2

```

Some reasons:
  * we would like the info to appear somewhere prominent 
  * we don't want it too far away from the MeshCoords
  * but we **don't** want it to break up the dimensions map..
  * So, I'm making a 'Mesh' section which _belongs_ in the dimensions-map ..
  * Otherwise, if we have CellMeasures/AuxCoords etc, it can really only go _after_ those 


**_Opinions ?!?_**

**Comment by wjbenfold** [comment]
I'd put it out of the coordinates - it's not a coordinate and I'd find it confusing to have it in there breaking up the coordinates.

That said, I'd far rather this happened in some way and then we can tweak it if we want it moving when we have more mesh users, rather than people spending ages discussing where it perfectly goes and it never getting implemented, so I'm happy with wherever the person who picks up this ticket decides to put it

**Comment by pp-mo** [comment]
At the risk of turning this into a discussion thread
 -- and probably a contentious one at that ...

To provide this info without a significant re-think to the existing ["cube summary object"](https://github.com/SciTools/iris/blob/main/lib/iris/_representation/cube_summary.py#L234) and ["cube printout"](https://github.com/SciTools/iris/blob/main/lib/iris/_representation/cube_printout.py#L131) implementations, I think the mesh/location info can only go into a new cube summary "section", which then must be either a 'vector' section (i.e. it participates in the dims-map), or a 'scalar' one.
I think anything would require us to invent a new _type_ of section, which is likely to have awkward knock-ons.
( e.g. the [notebook _html_repr_ output](https://github.com/SciTools/iris/blob/main/lib/iris/experimental/representation.py#L16) : though IIRC that may not work properly for unstructured cubes yet, anyway ?)

So, I wrote a nasty hacky bit of code to populate a cube with all possible sections.
Don't ask, but the outcome prints out like this :
```
air_potential_temperature / (K)     (time: 6; model_level_number: 70; -- : 100)
    Dimension coordinates:
        time                             x                      -        -
        model_level_number               -                      x        -
    Mesh coordinates:
        grid_latitude                    -                      -        x
        grid_longitude                   -                      -        x
    Auxiliary coordinates:
        level_height                     -                      x        -
        sigma                            -                      x        -
        surface_altitude                 -                      -        x
    Derived coordinates:
        altitude                         -                      x        x
    Cell measures:
        unknown                          -                      -        x
    Ancillary variables:
        vertical_angle                   -                      -        x
    Scalar coordinates:
        forecast_period             0.0 hours
        grid_latitude               -0.08325 degrees, bound=(-0.12825, -0.03825) degrees
    Cell methods:
        mean                        grid_latitude
    Attributes:
        source                      'Iris test case'
```

So, I the benefit of presenting it as a "vector" section, as above, is that you can put it alongside the mesh-coords within the dims map, and also highlight that it links to the mesh dimension.  But, as pointed out, it **_isn't_** a coord or any sort of 'dimensional metadata' component, so it doesn't really belong there.

However,  if we provide a "scalar" Mesh section, then the awkward part is that it may be some way away from the primary coordinate info in the dims map.  The best we can do is put it as the _first_ scalar section.
In the above case, that probably looks something like this :
```
air_potential_temperature / (K)     (time: 6; model_level_number: 70; -- : 100)
    Dimension coordinates:
        time                             x                      -        -
        model_level_number               -                      x        -
    Mesh coordinates:
        grid_latitude                    -                      -        x
        grid_longitude                   -                      -        x
    Auxiliary coordinates:
        level_height                     -                      x        -
        sigma                            -                      x        -
        surface_altitude                 -                      -        x
    Derived coordinates:
        altitude                         -                      x        x
    Cell measures:
        unknown                          -                      -        x
    Ancillary variables:
        vertical_angle                   -                      -        x
    Mesh:
        var_name                    Mesh2d
        location                    face
    Scalar coordinates:
        forecast_period             0.0 hours
        grid_latitude               -0.08325 degrees, bound=(-0.12825, -0.03825) degrees
    Cell methods:
        mean                        grid_latitude
    Attributes:
        source                      'Iris test case'
```

**Comment by trexfeathers** [comment]
Looks good, @pp-mo.

>it may be some way away from the primary coordinate info in the dims map

Having a `Mesh:` section stands out enough for me, and for dim-specific stuff we have the `Mesh coordinates:`.

**Comment by wjbenfold** [comment]
closed by #4778
